Add notes on datetime (#7468)
pingcap/docs: functions-and-operators/date-and-time-functions.md

TiDB supports all of the [date and time functions](https://dev.mysql.com/doc/refman/5.7/en/numeric-functions.html) available in MySQL 5.7.

> **Note:**
>

> > MySQL will often accept incorrectly formatted date and time values. For example, `'2020-01-01\n\t01:01:01'` and `'2020-01_01\n\t01:01'` are treated as valid date and time values.
> > TiDB makes a best-effort to match MySQL's behavior, but it might not match in all instances. It is recommended to correctly format dates, as the intended behavior for incorrectly formatted values is not documented, and is often inconsistent.

Reviewer: ```suggestion
> - MySQL will often accept the incorrectly formatted date and time values. For example, `'2020-01-01\n\t01:01:01'` and `'2020-01_01\n\t01:01'` are treated as valid date and time values.
> - TiDB makes the best effort to match MySQL's behavior, but it might not match in all instances. It is recommended to correctly format dates, because the intended behavior for incorrectly formatted values is not documented, and is often inconsistent.
```

> > TiDB makes a best-effort to match MySQL's behavior, but might not be correct in all instances. It is recommended to correctly format dates, as the intended behavior for incorrectly formated values is not documented, and is often inconsistent.

Reviewer 2: ```suggestion
> TiDB makes a best-effort to match MySQL's behavior, but it might not match in all instances. It is recommended to correctly format dates, as the intended behavior for incorrectly formatted values is not documented, and is often inconsistent.
```

For details, see [Date and Time Functions](https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html).

## MySQL compatibility

> The function `str_to_date()` is supported by TiDB, but is unable to parse all date and time values. In addition, the following date and time formatting options are **not supported**:

Reviewer 2: ```suggestion
The function `str_to_date()` is supported by TiDB, but is unable to parse all date and time values. In addition, the following date and time formatting options are **not implemented**:
```
